bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java

```diff
@@ -241,6 +320,23 @@ private boolean updateTitle() {
         return false;
     }
 
+    private void initializeIssuePRMap() {
+        // When bot restarts, the issuePRMap needs to get updated with this pr
```

Comment:
I'm not sure about this comment. This mapping needs to be updated every time there is a change to the associations, or if the map is currently missing associations for this PR. I don't think it makes sense to maintain an `initializedPRs` map, we should just always update the map.

This update process needs to be synchronized. The map is a `ConcurrentHashMap`, which is good, but we need to synchronize on the list value before trying to manipulate it.

Reply:
I make the bot maintaining the` initializedPRs` map because I think there is no need to always update the map here. We  only need to update the map here when the bot restarts and it's the first time for the bot to see this pr. Otherwise, the update is extra work. 

The real update for the map located in `CheckRun#getStatusMessage` from line 744 to 768. 

And yes, I need to synchronize on the list value before trying to manipulate it. Thanks for catching!

Reply:
Ah, I missed that part, that's good. With multiple locations doing updates to the map, I would recommend adding methods `addIssuePrMapping` and `removeIssuePrMapping` to `PullRequestBot` so you can encapsulate the dealing with lists and synchronization.